src/Database/Model.php

```diff
@@ -849,12 +849,10 @@ public function newPivot(EloquentModel $parent, array $attributes, $table, $exis
      */
     public function newRelationPivot($relationName, $parent, $attributes, $table, $exists)
     {
-        $definition = $this->getRelationDefinition($relationName);
+        $relation = $this->{$relationName}();
+        $pivotModel = $relation->getPivotClass();
 
-        if (!is_null($definition) && array_key_exists('pivotModel', $definition)) {
-            $pivotModel = $definition['pivotModel'];
-            return $pivotModel::fromAttributes($parent, $attributes, $table, $exists);
-        }
+        return $pivotModel::fromAttributes($parent, $attributes, $table, $exists);
```

Comment:
```suggestion
        return $pivotModel::fromRawAttributes($parent, $attributes, $table, $exists);
```

See https://github.com/wintercms/storm/commit/339af87267d355a8acef6dafbed9d2f2523099d2